src/librustdoc/html/render.rs

```diff
@@ -143,6 +144,55 @@ impl Impl {
     }
 }
 
+#[derive(Debug)]
+pub struct Error {
+    filename: Option<String>,
+    error: io::Error,
+}
+
+impl error::Error for Error {
+    fn description(&self) -> &str {
+        self.error.description()
+    }
+}
+
+impl Display for Error {
+    fn fmt(&self, f: &mut Formatter) -> fmt::Result {
+        match self.filename {
+            Some(ref file) => write!(f, "\"{}\": {}", file, self.error),
+            None => self.error.fmt(f),
+        }
+    }
+}
+
+impl Error {
+    pub fn from(e: io::Error) -> Error {
```

Comment:
It's not possible to do this (take a look at build_index and write_shared functions). I'll add it as much places as possible but not all.